frontend/packages/core/src/Network/errors.ts

```diff
@@ -0,0 +1,319 @@
+ * Contains metadata about the request that can be attached when filing a bug
+ * or providing other forms of feedback.
+ */
+export interface RequestInfo {
+  /**
+   * An opaque string that should only be interpreted by the service generating
+   * it.
+   *
+   * For example, it can be used to identify requests in the service's logs.
+   */
+  requestId?: string;
+  /**
+   * Any data that was used to serve this request.
+   *
+   * For example, an encrypted stack trace that can be sent back to the service
+   * provider for debugging.
+   */
+  servingData?: string;
+}
+
+/** Describes the resource that is being accessed. */
+export interface ResourceInfo {
+  /**
+   * A name for the type of resource being accessed, e.g. "sql table",
+   * "cloud storage bucket", "file", "Google calendar"; or the type URL
+   * of the resource: e.g. "type.googleapis.com/google.pubsub.v1.Topic".
+   */
+  resourceType?: string;
+  /**
+   * The name of the resource being accessed. For example, a shared calendar
+   * name: "example.com_4fghdhgsrgh@group.calendar.google.com", if the current
+   * error is [google.rpc.Code.PERMISSION_DENIED][google.rpc.Code.PERMISSION_DENIED].
+   */
+  resourceName?: string;
+  /**
+   * The owner of the resource (optional).
+   *
+   * For example, "user:<owner email>" or "project:<Google developer project id>".
+   */
+  owner?: string;
+  /**
+   * Describes what error is encountered when accessing this resource.
+   *
+   * For example, updating a cloud project may require the `writer` permission
+   * on the developer console project.
+   */
+  description?: string;
+}
+
+/**
+ * Provides links to documentation or for performing an out of band action.
+ *
+ * For example, if a quota check failed with an error indicating the calling
+ * project hasn't enabled the accessed service, this can contain a URL pointing
+ * directly to the right place in the developer console to flip the bit.
+ */
+export interface Help {
+  /** URL(s) pointing to additional information on handling the current error. */
+  links?: {
+    /** Describes what the link offers. */
+    description?: string;
+    /**
+     * The URL of the link.
+     */
+    url?: string;
+  }[];
+}
+
+/**
+ * Provides a generic interface for error details. This should only be used if
+ * the error type cannot be mapped to a more specific type.
+ */
+export interface Details {
+  [key: string]: string;
+}
+
+/** An RPC error that occurred on the server. */
+export interface GrpcError extends Error {
+  /** The status code. */
+  code?: number;
+  /**
+   * A developer-facing error message in English. Any user-facing error messages
+   * will be sent in the details field.
+   */
+  message: string;
+  /** A list of objects that carry error details from the server. */
+  details?: (
+    | RetryInfo
+    | DebugInfo
+    | QuotaFailure
+    | ErrorInfo
+    | PreconditionFailure
+    | BadRequest
+    | RequestInfo
+    | ResourceInfo
+    | Help
+    | Details
+  )[];
```

Comment:
shouldn't this just be the types that we have renderers for and `any`? it's impossible to maintain a full list of possible details objects, it could be any proto object.

Reply:
👍  on the any type (I can swap it out for the generic details interface). It is impossible to account for them all and we could most certainly reduce these but I figured having a starting set would provide the most flexibility if other users wanted to utilize these types in their workflows. Ideally we would just consume a googleapis library that already constructs these from the proto definitions but I could not find one that exists.